test/e2e/apimachinery/flowcontrol.go

```diff
@@ -51,21 +51,26 @@ var _ = SIGDescribe("API priority and fairness", func() {
 		matchingUsername := "noxu"
 		nonMatchingUsername := "foo"
 
-		ginkgo.By("creating a testing prioritylevel")
+		ginkgo.By("creating a testing priority-level")
 		createdPriorityLevel, cleanup := createPriorityLevel(f, testingPriorityLevelName, 1)
 		defer cleanup()
 
-		ginkgo.By("creating a testing flowschema")
+		ginkgo.By("creating a testing flow-schema")
 		createdFlowSchema, cleanup := createFlowSchema(f, testingFlowSchemaName, 1000, testingPriorityLevelName, []string{matchingUsername})
 		defer cleanup()
 
-		ginkgo.By("checking response headers contain flow-schema/priority-level uid")
-		if !testResponseHeaderMatches(f, matchingUsername, string(createdPriorityLevel.UID), string(createdFlowSchema.UID)) {
-			framework.Failf("matching user doesnt received UID for the testing priority-level and flow-schema")
-		}
-		if testResponseHeaderMatches(f, nonMatchingUsername, string(createdPriorityLevel.UID), string(createdPriorityLevel.UID)) {
-			framework.Failf("non-matching user unexpectedly received UID for the testing priority-level and flow-schema")
-		}
+		ginkgo.By("retrieving catch-all flow-schema and priority-level objects")
+		globalDefaultFlowSchema := getFlowSchema(f, "global-default")
+		globalDefaultPriorityLevel := getPriorityLevelConfig(f, globalDefaultFlowSchema.Spec.PriorityLevelConfiguration.Name)
+
+		var response *http.Response
+		ginkgo.By("checking response headers contain flow-schema and priority-level UID for a matching user")
+		response = makeRequest(f, matchingUsername)
+		verifyResponseHeaderMatch(response, string(createdPriorityLevel.UID), string(createdFlowSchema.UID))
+
+		ginkgo.By("checking response headers contain catch-all flow-schema and priority-level UID for a non-matching user")
+		response = makeRequest(f, nonMatchingUsername)
+		verifyResponseHeaderMatch(response, string(globalDefaultPriorityLevel.UID), string(globalDefaultFlowSchema.UID))
```

Comment:
I do not recommend expecting any particular classification for a non-matching request.  That is a matter of configuation that this test is not controlling.  We could test that the relevant headers are present but should not insist on them identifying any particular flow schema or priority level.

Reply:
sounds good to me.